test/integration/provider-mock.ts

```diff
@@ -0,0 +1,415 @@
+import cors = require("cors");
+import express = require("express");
+import q = require("q");
+import bodyParser = require("body-parser");
+import basicAuth = require("basic-auth");
+import * as http from "http";
+
+export default (port: number): q.Promise<http.Server> => {
+	const server: express.Express = express();
+	server.use(cors());
+	server.use(bodyParser.json());
+	server.use(bodyParser.urlencoded({
+		extended: true
+	}));
+
+	let stateData = "";
+
+	server.get("/", (req, res) => {
+		res.json({
+			greeting: "Hello"
+		});
+	});
+
+	server.get("/fail", (req, res) => {
+		res.json({
+			greeting: "Oh noes!"
+		});
+	});
+
+	server.get("/provider-states", (req, res) => {
+		res.json({
+			me: ["There is a greeting"],
+			anotherclient: ["There is a greeting"]
+		});
+	});
+
+	server.post("/provider-state", (req, res) => {
+		stateData = "State data!";
+		res.json({
+			greeting: stateData
+		});
+	});
+
+	server.get("/somestate", (req, res) => {
+		res.json({
+			greeting: stateData
+		});
+	});
+
+	server.post("/", (req, res) => {
+		res.json({
+			greeting: `Hello ${req.body.name}`
+		});
+	});
+
+	server.get("/contract/:name", (req, res) => {
+		const options = {
+			root: __dirname,
+			dotfiles: "deny",
+			headers: {
+				"x-timestamp": Date.now(),
+				"x-sent": true
+			}
+		};
+
+		const fileName = req.params.name;
+		res.sendFile(fileName, options, (err) => {
+			if (err) {
+				console.log(err);
+				res.status(500).end();
+			} else {
+				console.log("Sent:", fileName);
+			}
+		});
+	});
+
+// Pretend to be a Pact Broker (https://github.com/bethesque/pact_broker) for integration tests
+// Let"s add Auth for good measure
+	const auth = (req, res, next) => {
+		const user = basicAuth(req);
+		if (user && user.name === "foo" && user.pass === "bar") {
+			return next();
+		} else {
+			res.set("WWW-Authenticate", "Basic realm=Authorization Required");
+			return res.sendStatus(401);
+		}
+	};
+
+// Verification result
+	server.post("/pacts/provider/:provider/consumer/:consumer/pact-version/:version/verification-results", (req, res) => res.json({}));
+
+	server.get("/pacts/provider/they/consumer/me/latest", auth, (req, res) => {
+		res.json({
+			"consumer": {
```

Comment:
Wondering if we should pull these `.json` bits into separate files and server them up to make things more readable?

Reply:
Yeah, not a bad idea.